mobile/lib/utils/photo_manager_util.dart

```diff
@@ -10,3 +13,128 @@ Future<PermissionState> requestPhotoMangerPermissions() {
     ),
   );
 }
+
+final _logger = Logger("PhotoManagerUtil");
+// This is a wrapper for safe handling of PhotoManager.startChangeNotify() and
+// PhotoManager.stopChangeNotify(). Since PhotoManager is globally shared, we want
+// to make sure no notification is sent while it should not. The logic is that it will
+// only start if no other asset (or task) requested to stop changes, or if the expiration
+// time for the asset (task) expired. '_processingAssets' should be seen as a queue of
+// open requests.
+class PhotoManagerSafe {
```

Comment:
Sorry, but to me this feels like additional complexity that we like prefer to avoid adding.
Is it possible to simplify this part either via debouncer or leveraging the fact that all calls to PhotoManager start/stop happens on the same thread?

Reply:
I agree, actually it is not even needed with the Local Service Lock (I tested it, but it was already making sense). I have removed it at the moment as it is trying to solve a bigger issue IMO.